aea/protocols/dialogue/base.py

```diff
@@ -371,6 +371,7 @@ def __eq__(self, other) -> bool:
             and self.message_class == other.message_class
             and self._incoming_messages == other._incoming_messages
             and self._outgoing_messages == other._outgoing_messages
+            and self._ordered_message_ids == other._ordered_message_ids
```

Comment:
probably redundant here, cause incoming and outgoing messages will check ids and targets that reflects order of the messages

Reply:
Disagree, we don't have total order from only incoming and outgoing messages lists.